src/Microsoft.Azure.WebJobs.Host/Scale/ScaleLoggerExtension.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Host.Scale
+{
+    public static class ScaleLoggerExtension
```

Comment:
Consider using high-performance logging which will pre-compile the template.
https://learn.microsoft.com/en-us/dotnet/core/extensions/high-performance-logging